### PR TITLE
14 Added Implementation for self-trained model

### PR DESCRIPTION
**Pull request by craftamap**
This implements the self-trained smile detector. More on the model after I published it.

### CODE REVIEW

**Review by AnnaGsell — DISMISSED** [review]

File: app/src/main/java/de/hsaugsburg/teamulster/sohappy/fragment/CameraFragment.kt [hunk 1]
```diff
@@ -132,7 +132,7 @@ class CameraFragment: Fragment() {
         requestCode: Int, permissions: Array<String?>, grantResults: IntArray
     ) {
         if (requestCode == REQUEST_CODE_PERMISSIONS) {
-            startCameraIfReady()
+            //startCameraIfReady()
```
Comment (limkokyew) [inline app/src/main/java/de/hsaugsburg/teamulster/sohappy/fragment/CameraFragment.kt:135]: I'd suggest to either remove or uncomment this line.

File: app/src/main/java/de/hsaugsburg/teamulster/sohappy/analyzer/detector/smiledetectorimpl/SoHappyTFLiteSmileDetector.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package de.hsaugsburg.teamulster.sohappy.analyzer.detector.smiledetectorimpl
+
+import android.app.Activity
+import android.graphics.Bitmap
+import de.hsaugsburg.teamulster.sohappy.analyzer.BitmapEditor.convertToByteBuffer
+import de.hsaugsburg.teamulster.sohappy.analyzer.detector.SmileDetector
+import java.nio.ByteBuffer
+
+class SoHappyTFLiteSmileDetector(activity: Activity) :
```
Comment (limkokyew) [inline app/src/main/java/de/hsaugsburg/teamulster/sohappy/analyzer/detector/smiledetectorimpl/SoHappyTFLiteSmileDetector.kt:9]: KDoc is missing here.

**Review by limkokyew — COMMENTED** [review]
Two very minor issues, but other than that - good work!

File: app/src/main/java/de/hsaugsburg/teamulster/sohappy/analyzer/detector/smiledetectorimpl/SoHappyTFLiteSmileDetector.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+     *
+     * @param bitmap the image which TF lite will be running detection on.
+     * @return Companion.SmileDetectionResult
+     * */
+    override fun detect(img: Bitmap): SmileDetector.Companion.SmileDetectionResult? {
+        val predictionResults: ArrayList<Companion.Recognition> = super.execute(img)
+        val firstPredictionResult = predictionResults[0]
+        var isSmiling = false
+        // if Happy is detected, set isSmiling = true
+        // TODO: This has to be adapted depending on each model
```
Comment (mac641) [inline app/src/main/java/de/hsaugsburg/teamulster/sohappy/analyzer/detector/smiledetectorimpl/SoHappyTFLiteSmileDetector.kt:46]: This TODO can be removed I guess 🤔

**Review by mac641 — CHANGES_REQUESTED** [review]

**Review by diehmlukas — DISMISSED** [review]

**Review by AnnaGsell — DISMISSED** [review]

**Review by mac641 — APPROVED** [review]

**Pull request merged**